### PR TITLE
Simplify insertion of vcpkg-tool release

### PR DESCRIPTION
**Pull request by michaelbprice**
This pull request has 3 goals.

1) Reduce the number of files and source locations where changes are required when inserting a new vcpkg-tool release.

Currently, the information is spread across 2 bootstrap scripts. In the Unix bootstrap script, the changes span about 20 lines of code. Consolidating the data into a single-purpose file reduces the chances for mistakes and also allows for better isolation between tool insertion and improvements/fixes to the bootstrap scripts.

2) Enable automation of vcpkg-tool insertion.

Today, the step between the release tagging of vcpkg-tool and insertion into vcpkg is a manual one. Isolating the insertion data into a simply formatted file should make automating this step easier (e.g., an Actions workflow that updates the data file and opens a vcpkg PR).

3) Easier adoption and rollback of tool releases.

If a user updates vcpkg in order to gain access to a new port, but there is a bug in the inserted vcpkg-tool, the isolation of this data into a separate file provides an easy mechanism for temporarily rolling back to an unaffected tool version. Similarly, if a user wishes to try out a tagged vcpkg-tool release, but does not want to update their vcpkg registry, this allows them to easily experiment with the new tool release.

Incidentally, this also exposes the fact that the Windows bootstrap process does not currently compare checksums of downloaded binaries against expected values. The technique used here would fully support that improvement if we wanted to add it.

### CODE REVIEW

**Comment by michaelbprice** [comment]
This changes in this PR are not as flexible as those in https://github.com/microsoft/vcpkg/pull/32156, but it largely solves the same problem while also having other benefits. If this PR is merged, the other should be closed.

**Comment by michaelbprice** [comment]
Given https://github.com/microsoft/vcpkg/pull/32649 is in flight, this should only be merged after that tool insertion is complete (which would require updating the data in this PR to resolve the impending merge conflict).

**Comment by BillyONeal** [comment]
> Incidentally, this also exposes the fact that the Windows bootstrap process does not currently compare checksums of downloaded binaries against expected values.

This is intentional, on Windows we check the Authenticode signature. See:

https://github.com/microsoft/vcpkg-tool/blob/6040912b8dc4c0725b20510cfb2c5c79f35da168/src/tls12-download.c#L370-L389

File: .vcpkg-tool [hunk 1]
```diff
@@ -0,0 +1,7 @@
+VCPKG_TOOL_RELEASE_TAG=2023-06-22
```
Comment (BillyONeal) [inline .vcpkg-tool:1]: I think that this file should be `scripts/vcpkg-tool-release.txt` or similar.

In particular it should not be `.hidden`.

File: .vcpkg-tool [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+VCPKG_EXE_SHA=
+VCPKG_ARM64_SHA=
```
Comment (BillyONeal) [inline .vcpkg-tool:3]:
```suggestion
```

File: scripts/bootstrap.ps1 [hunk 3]
```diff
@@ -45,7 +45,10 @@ while (!($vcpkgRootDir -eq "") -and !(Test-Path "$vcpkgRootDir\.vcpkg-root"))
 
 Write-Verbose "Examining $vcpkgRootDir for .vcpkg-root - Found"
 
-$versionDate = '2023-06-22'
+# Read the vcpkg-tool config file to determine what release to download
+$CONFIG = ConvertFrom-StringData ([io.file]::ReadAllText("$vcpkgRootDir/.vcpkg-tool"))
+$versionDate = $CONFIG.VCPKG_TOOL_RELEASE_TAG
```
Comment (BillyONeal) [inline scripts/bootstrap.ps1:50]:
```suggestion
$Config = ConvertFrom-StringData (Get-Content "$vcpkgRootDir/.vcpkg-tool")
$versionDate = $Config.VCPKG_TOOL_RELEASE_TAG
```

**Review by BillyONeal — COMMENTED** [review]
TIL of `ConvertFrom-StringData`, that's nice!

I like this direction other than the nitpicks.

**Comment by Cheney-W** [comment]
Moving the PR to draft. Please set it to "ready for review" once changes have been applied.

**Review by BillyONeal — APPROVED** [review]

**Review by vicroms — APPROVED** [review]

File: scripts/bootstrap.ps1 (hunk 3 above)
Comment (michaelbprice) [reply scripts/bootstrap.ps1]: I had tried `Get-Content` at first, but it had weird issues. I don't remember exactly, but I think it had to do with line endings and how much of the file it read. I suspect that the `-Raw` that you added in the most recent version addressed that issue. Much nicer than resorting to that `ReadAllText` method.

**Comment by michaelbprice** [comment]
LGTM

**Pull request merged**